docs/logic.md

```diff
@@ -6,34 +6,36 @@ A controller handles the following events if they have "Completed" status.
 
 Other events are ignored. <br />
 
-In case of an exception, the processing of event will be restarted, except the creation event which has no entity in CloudStack (see the first paragraph above). <br />
+In case of an exception, the processing of event will be restarted, except the creation event which has no entity (vm or account) in CloudStack. <br />
 
 
 **Event processing at the service interaction level.** <br />
 
 The processing of VM creation event consists of the following steps: <br />
-  * a) get the appropriate account ID via CloudStackService <br />
-  * b) try to retrieve tokens from zookeeper. Token can have 'RO' or 'RW' policy <br />
-  * c) if the tokens don't exist: <br />
-    * 1) the tokens are created in vault <br />
-    * 2) zookeeper nodes are created for keeping VM tokens <br />
-    * 3) write the vault tokens to zookeeper (in case of an exception, the tokens will be revoked) <br />
-  * d) write the tokens in CloudStack as UserVM tags <br />
+  1. get the appropriate account ID via CloudStackService: <br />
+    1.1. If the account did not retrieve, skip other steps.
+  2. try to retrieve tokens from zookeeper. Token can have 'RO' or 'RW' policy <br />
+  3. if the tokens don't exist: <br />
+    3.1. the tokens are created in vault <br />
+    3.2. zookeeper nodes are created for keeping VM tokens <br />
+    3.3. write the vault tokens to zookeeper (in case of an exception, the tokens will be revoked) <br />
+  4. write the tokens in CloudStack as UserVM tags <br />
 
 The processing of account creation event consists of the following steps: <br />
-  * a) check the account existence <br />
-  * b) try to retrieve tokens from zookeeper. Token can have 'RO' or 'RW' policy <br />
-  * c) if the tokens don't exist: <br />
-    * 1) the tokens are created in vault <br />
-    * 2) zookeeper nodes are created for keeping Account tokens <br />
-    * 3) write the vault tokens to zookeeper (in case of an exception, the tokens will be revoked) <br />
-  * d) write the tokens in CloudStack as Account tags <br />
+  1. check the account existence <br />
+    1.1. If the account did not exist, skip other steps.
+  2. try to retrieve tokens from zookeeper. Token can have 'RO' or 'RW' policy <br />
+  3. if the tokens don't exist: <br />
```

Comment:
If the tokens exist go to the next step, otherwise:

Reply:
ok